server/internal/application/user/user.go

```diff
@@ -0,0 +1,62 @@
+package user
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	ErrNameNotValid = errors.New("pippo is not valid")
+)
+
+type User struct {
+	Username  string
+	Password  string
+	Name      string
+	Surname   string
+	Email     string
+	Role      string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type UserCreateDTO struct {
+	ID        string
+	Username  string
+	Password  string
+	Name      string
+	Surname   string
+	Email     string
+	Role      string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type UserUpdateDTO struct {
+	ID       string
+	Username *string
+	Password *string
+	Name     *string
+	Surname  *string
+	Email    *string
+	Role     *string
+}
+
+type UserRetrieveDTO struct {
+	ID        string
+	Username  string
+	Password  string
+	Name      string
+	Surname   string
+	Email     string
+	Role      string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type UserRepository interface {
+	Register(u UserCreateDTO) error
+	Update(u UserUpdateDTO) error
```

Comment:
We don't really need to differentiate between creation and update, a single `Persist(u User) error`  or `Upsert(u User) error` is sufficient for both cases (in SQL this can be easily implemented using a `INSERT ... ON CONFLICT (id) DO UPDATE` statement).